### ISSUE TITLE
Stream pulse audio to Sonos player?

### ISSUE THREAD

**Issue by eeeebbbbrrrr** [w4/sonos-cli#1]
Hey, cool project!

I know it’s been a few years since this was last updated and no biggie if you have no further plans, but…

Did you ever put thought into being able to stream a pulse audio device (or mp3 or whatever) to the Sonos network?  Is that possible at all?

I haven’t run across a protocol spec. Did you have to reverse what this does?

$480 for a Sonos port v/s my time to implement the same in software… 🤔 

Thanks for your time!

eric

**Comment by w4** [comment]
Hey Eric,

There's some light documentation dotted around the web such as for [AVTransport](https://svrooij.io/sonos-api-docs/services/av-transport.html), it's not the most in-depth but it's certainly a jumping off point for your travels into the depths.

I've not put much thought into Pulseaudio specifically, but a setup I used recently was streaming via icecast and picking up that m3u on the Sonos, latency wasn't the best however (though I imagine the whole Port business doesn't have much better latency itself, given the Play:5 line in latency).

Something with less hops would certainly be welcome in this household though, sounds like it could be a fun project - let me know if you break any ground on it!

Thanks,
Jordan

**Comment by eeeebbbbrrrr** [comment]
I ended up ordering a Sonos Port anyways for my record player. I changed my plans a bit with that. Should arrive later today.  

However!  I’ll look into this anyways.  I’ve got a Linux box whose sound card I’d like to get connected to to Sonos and I’m not willing to spend another five hundo on that. 

Thanks for the AVTransport link and your icecast experiment at least proves the idea is possible.  

I just wanna “cat /dev/audio | sonos-cli —stream” and let that bad boy run in the background. 

I’ll check back in after awhile. Thanks again!
